tests/unittest_scoped_nodes.py

```diff
@@ -2637,5 +2651,49 @@ def test_non_frame_node():
         assert module.body[1].value.locals["x"][0].frame(future=True) == module
 
 
+def test_deprecation_of_doc_attribute() -> None:
```

Comment:
We should add some more tests here. One block to make sure we don't unintentionally raise `DeprecationWarning`.
```py
    doc_node = nodes.Const("Docstring")
    with warnings.catch_warnings():
        # If 'doc' isn't passed to Module, ClassDef, FunctionDef,
        # no DeprecationWarning should be raised
        # Modify warnings filter to raise error for DeprecationWarning
        warnings.simplefilter("error", DeprecationWarning)
        node_module = nodes.Module(name="MyModule")
        node_module.postinit(body=[], doc_node=doc_node)
        assert node_module.doc_node == doc_node
        node_class = nodes.ClassDef(name="MyClass")
        node_class.postinit(bases=[], body=[], decorators=[], doc_node=doc_node)
        assert node_class.doc_node == doc_node
        node_func = nodes.FunctionDef(name="MyFunction")
        node_func.postinit(args=nodes.Arguments(), body=[], doc_node=doc_node)
        assert node_func.doc_node == doc_node
```

And another to make sure, the `doc` node is assigned even if only `doc_node` is passed to `postinit`. Otherwise we can't remove the old usage without breaking backwards compatibility. _This currently fails._
```py
    # Test 'doc' attribute if only 'doc_node' is passed
    with pytest.warns(DeprecationWarning) as records:
        assert node_module.doc == "Docstring"
        assert len(records) == 1
    with pytest.warns(DeprecationWarning) as records:
        assert node_class.doc == "Docstring"
        assert len(records) == 1
    with pytest.warns(DeprecationWarning) as records:
        assert node_func.doc == "Docstring"
        assert len(records) == 1
```

Reply:
Tests added!